sentry/sentry/nat_option.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2020-2022 The Silkworm Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "nat_option.hpp"
+#include <boost/algorithm/string/predicate.hpp>
+
+namespace silkworm::sentry {
+
+bool lexical_cast(const std::string& input, NatOption& value) {
+    if (input == "none") {
+        value = {};
+        return true;
+    }
+    if (boost::algorithm::istarts_with(input, "extip:")) {
+        auto ip_str = input.c_str() + 6;
```

Comment:
I'd rather 

```suggestion
        const auto ip_str{&input[6]};
```

Reply:
I did it this way, because `data()` is not guaranteed to be null-terminated. Isn't `&input[6]` is a pointer inside data? `operator[]` docs says nothing about null-termination.

I've noticed that on macOS `data()` and `c_str()` are practically the same, but I was not sure about the other platforms, so I decided to stay safe.

Reply:
It is the same.
This is implementation of `c_str()` which simply returns the base pointer to data
```
return _Mypair._Myval2._Myptr();
```
This is the implementation of `data()`
```
return _Mypair._Myval2._Myptr();
```
As you see are identical and carry the same assertions.

Random access operator [] is implemented like this
```
return _Mypair._Myval2._Myptr()[_Off];
```

Hence what proposed is more concise and more C++ style.

Reply:
Besides we have the style consraint to brace initialize vars
so 
`auto ip_str = input.c_str() + 6;`
has to be corrected with
`auto ip_str {input.c_str() + 6};`

Reply:
Since C++11 data is **always** guaranteed to be null terminated.